### PR TITLE
Award points for null-safety

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

**Comment by sigurdm** [comment]
cc @mit-mit

**Review by jonasfj — APPROVED** [review]

File: lib/src/create_report.dart [hunk 1]
```diff
@@ -898,7 +898,7 @@ _Issue _explanationToIssue(Explanation explanation) =>
 Future<ReportSection> _nullSafety(String packageDir, Pubspec pubspec) async {
   // TODO(sigurdm): Currently we don't give any points for null-safety.
```
Comment (mit-mit) [inline lib/src/create_report.dart:899]: Remove TODO?

File: lib/src/create_report.dart [hunk 2]
```diff
@@ -911,18 +911,14 @@ Future<ReportSection> _nullSafety(String packageDir, Pubspec pubspec) async {
     if (pubspec.sdkConstraintStatus.hasOptedIntoNullSafety) {
       if (nullSafetyTags.contains('is:null-safe')) {
         subsection = _Subsection(
-            'Package and dependencies are fully migrated to null-safety, '
-            'and will be awarded additional points in a planned future revision '
-            'of the pub.dev points model.',
+            'Package and dependencies are fully migrated to null-safety!',
```
Comment (mit-mit) [inline lib/src/create_report.dart:914]: No hyphen in `null-safety` (generally, not just here)

**Review by isoos — APPROVED** [review]
please also update CHANGELOG

File: test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.0.json [hunk 3]
```diff
@@ -85,9 +85,9 @@
...
         "grantedPoints": 0,
-        "maxPoints": 0,
+        "maxPoints": 20,
         "status": "failed",
-        "summary": "### [x] 0/0 points: Package declares support for null-safety, but there are issues.\n\nPackages with full null-safety support will be awarded additional points in a planned future revision of the pub.dev points model.\n\n<details>\n<summary>\nPackage is not null safe\n</summary>\n\nBecause:\n* `_dummy_pkg` that depends on:\n* `retry` that doesn't opt in to null safety\n</details>\n<details>\n<summary>\nFor more information\n</summary>\n\nTry running `pub outdated --mode=null-safety`.\nBe sure to read the [migration guide](https://dart.dev/null-safety/migration-guide).\n</details>"
+        "summary": "### [x] 0/20 points: Package declares support for null-safety, but there are issues.\n\n<details>\n<summary>\nPackage is not null safe\n</summary>\n\nBecause:\n* `_dummy_pkg` that depends on:\n* `retry` that doesn't opt in to null safety\n</details>\n<details>\n<summary>\nFor more information\n</summary>\n\nTry running `pub outdated --mode=null-safety`.\nBe sure to read the [migration guide](https://dart.dev/null-safety/migration-guide).\n</details>"
```
Comment (mit-mit) [inline test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.0.json:90]: Maybe change "Package declares support for null-safety, but there are issues" to "Null safety support has one or more issues"?

**Review by mit-mit — CHANGES_REQUESTED** [review]

File: lib/src/create_report.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/create_report.dart]: Done

File: lib/src/create_report.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/create_report.dart]: Thanks! Done

File: test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.0.json (hunk 3 above)
Comment (sigurdm) [reply test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.0.json]: Done

**Review by sigurdm — COMMENTED** [review]
Added a Changelog entry

**Review by mit-mit — APPROVED** [review]

**Pull request merged**